app/assets/javascripts/models/data/service_options.js.coffee

```diff
@@ -176,14 +176,13 @@ ns.ServiceOptions = do (ko, edsc = @edsc, KnockoutModel = @edsc.models.KnockoutM
       return true if @granuleAccessOptions().methods?.length == 0
       
       result = false
-      selected = false
       for m in @accessMethod()
         if @granuleAccessOptions().methods?.length == 1
           m.method(m.availableMethods[0].name)
-        result = true if m.method()? && m.isValid()
+        result = true if m.isValid()
```

Comment:
This changes the logic for determining whether or not result is true. Previously it was `result = true if m.method()? && m.isValid()` which requires both of those conditions to be true. This change has modified this logic to be `OR` by setting them separately. Your update shouldnt require any change to this logic, you're just ensuring that if there is only one data access method, that its pre-checked for the user. 

Again, I don't think any of this is necessary. Line 180 and 181 are the only updated needed to accomplish your goal unless I'm completely missing something.

Reply:
@abbottry - There are two goals :)

Lines 180 and 181 do the first goal - which is preselect a method if there is only one method - the first goal.

The second goal is whether or not the 'submit' button is enabled.  Changing the logic to effectively being an 'or' is required for this to work - the submit button should not be enabled if a method is not selected.  Therefore, result = false if !m.method()? makes sure that it returns false - thus disabling the 'submit' button - in the case that the method has not been chosen.